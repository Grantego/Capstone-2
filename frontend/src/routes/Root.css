.container-div {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    margin-top: 200px;
}


#root {
    background-color: #e6a64c4b;
}

#login-btn {
    margin-left: 20px;
}

#register-btn {
    margin-left: 20px;
}

#chord-details {
    height: 100%;
    width: 100%;
    display: flex;
    align-items: center;
    flex-direction: column;
}

#chord-detail-name {
    font-size: 5vw;
    margin-top: 5vw;
    margin-bottom: 3vw;
}

#chord-detail-spelling {
    font-size: 3vw;
}

#chord-chart-box {
    display: flex;
    width: 45vw;
    height: 50vw;
    margin-top: 10rem;
    padding: 20px;
    justify-content: center;
    background-color: rgb(238, 226, 210);
    box-shadow: 2px 2px 5px black;
    border: 1vw solid #e6a64c;
    border-radius: 8px;
}

#chord-detail-img {
    max-width: 100%;
    max-height: 100%;
    object-fit: contain;
}
a {
    color: black; /* Set the link color to black */
    text-decoration: none; /* Remove underline */
}


@media (max-width: 600px) {

    #chord-detail-name {
        font-size: 10vw;
        margin-top: 5vw;
        margin-bottom: 3vw;
    }
    
    #chord-detail-spelling {
        font-size: 6vw;
    }
    
    #chord-chart-box {
        margin-top: 20vw;
        width: 60vw;
        height: 70vw;
    }

  }