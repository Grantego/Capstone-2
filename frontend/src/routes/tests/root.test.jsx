import React from 'react';
import { render, screen } from '@testing-library/react';
import Root from '../root';
import { BrowserRouter } from 'react-router-dom';
import { describe, it, beforeEach, expect } from 'vitest';

describe('Root component', () => {
  beforeEach(() => {
    // Clear all instances and calls to constructor and all methods:
    localStorage.clear();
  });

  it("matches snapshot", function() {
    const {asFragment} = render(<BrowserRouter><Root/></BrowserRouter>)
    expect(asFragment()).toMatchSnapshot();
  })

  it("renders without crashing", () => {
    render(<BrowserRouter><Root/></BrowserRouter>)
  })

  it('displays "Chord Assistant" when user is not logged in', () => {
    render(<BrowserRouter><Root/></BrowserRouter>);
    expect(screen.getByText(/chord assistant/i)).toBeInTheDocument();
  });

  it('displays navbar', () => {
    render(<BrowserRouter><Root/></BrowserRouter>)
    expect(screen.getByText("Chord Helper")).toBeInTheDocument()
  })

  it('displays "Welcome back, [username]" when user is logged in', () => {
    const username = 'testuser';
    localStorage.setItem('token', 'sampletoken');
    localStorage.setItem('username', username);
    
    render(<BrowserRouter><Root /></BrowserRouter>);
    expect(screen.getByText(`Welcome back, ${username}`)).toBeInTheDocument();
  });
});