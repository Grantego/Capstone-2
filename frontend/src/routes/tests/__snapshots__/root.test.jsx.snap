// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root component > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-light bg-faded"
  >
    <div
      class="container-fluid"
    >
      <a
        class="me-auto navbar-brand"
        href="/"
      >
        Chord Helper
      </a>
      <button
        aria-label="Toggle navigation"
        class="me-2 navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/signup"
            >
              Sign Up
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class=""
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    class="container-div"
  >
    <h1>
      Chord Assistant
    </h1>
  </div>
</DocumentFragment>
`;
